### PR TITLE
 add eds service_name to LoadStats proto and populate it in LRS request

### PR DESCRIPTION
**Pull request by stevenzzzz**
For an explanation of how to fill out the fields, please see the relevant section 
in [PULL_REQUESTS.md](./PULL_REQUESTS.md)

*Description*:
   Add eds_cluster_service_name in load report proto. 
   This new field gives LRS management server the  extra info that two clusters may be using the same set of hosts to serve, and it's the management server's call whether to aggregate on the clusterStats.

*Risk Level*: LOW (new field)
*Testing*: unit test
*Release Notes*:
[Optional Fixes #Issue] #5986

### CODE REVIEW

**Comment by stevenzzzz** [comment]
@htuch PTAL

**Comment by htuch** [comment]
@stevenzzzz can you merge master to pick up the `compile_time_options` CI fix?

**Comment by stevenzzzz** [comment]
sync done, thanks!

File: source/common/upstream/upstream_impl.h [hunk 1]
```diff
@@ -508,6 +508,8 @@ class ClusterInfoImpl : public ClusterInfo {
 
   bool drainConnectionsOnHostRemoval() const override { return drain_connections_on_host_removal_; }
 
+  absl::optional<std::string> eds_service_name() const override { return eds_service_name_; }
```
Comment (htuch) [inline source/common/upstream/upstream_impl.h]: I think `eds_service_name` is fine here.

File: test/common/upstream/load_stats_reporter_test.cc [hunk 2]
```diff
@@ -117,6 +117,7 @@ TEST_F(LoadStatsReporterTest, ExistingClusters) {
   // Start reporting on foo.
   NiceMock<MockClusterMockPrioritySet> foo_cluster;
```
Comment (htuch) [inline test/common/upstream/load_stats_reporter_test.cc]: You'll also want to write a case for `load_stats_integration_test` to validate e2e.

Comment (stevenzzzz) [reply test/common/upstream/load_stats_reporter_test.cc]: added.

File: source/common/upstream/upstream_impl.h (hunk 1 above)
Comment (stevenzzzz) [reply source/common/upstream/upstream_impl.h]: I am not sure I understand you here. Are you talking about the name or the return value type?

**Comment by stevenzzzz** [comment]
oops, tried to compress load git commits but compressed the pushed version. 
PR updated. Added some new tests in upstream_impl_test.

File: source/common/upstream/upstream_impl.h (hunk 1 above)
Comment (htuch) [reply source/common/upstream/upstream_impl.h]: I was just pointing out that it's fine to use this internally, but not really the best choice in API. Resolved.

File: api/envoy/api/v2/endpoint/load_report.proto [hunk 3]
```diff
@@ -117,10 +117,16 @@ message EndpointLoadMetricStats {
...
 // [#not-implemented-hide:] Not configuration. TBD how to doc proto APIs.
+// Next ID: 7
 message ClusterStats {
   // The name of the cluster.
   string cluster_name = 1 [(validate.rules).string.min_bytes = 1];
 
+  // The eds_cluster_config service_name of the cluster.
+  // It's possible that two clusters send the same service_name to EDS,
+  // in that case, the management server is supposed to do aggregation on the load reports.
+  string cluster_eds_service_name = 6;
```
Comment (htuch) [inline api/envoy/api/v2/endpoint/load_report.proto]: Hmm, my original comment was lost I think. In any case I think this should just be `service_name`, for consistency with the EDS declaration in existing CDS. I.e. `cluster_service_name`.

Comment (stevenzzzz) [reply api/envoy/api/v2/endpoint/load_report.proto]: heh, I erased the previous commit by a "force push" , which I hope was just "folding local new commits" into a "new commit".
I don't have strong feeling about the name, my idea is the eds_cluster_service_name tells where this name "comes from" (cluster::eds_cluster_config::service_name), whereas any a single service_name may not convey that notion.

Updated.

**Pull request merged**